content/blog/2023/01/camel4roadmap/index.md

```diff
@@ -0,0 +1,76 @@
+---
+title: "Roadmap to Camel 4"
+date: 2023-01-03
+draft: false
+authors: [davsclaus, oscerd, essobedo]
+categories: ["Roadmap"]
+preview: "Roadmap to Apache Camel 4 and impact on Camel 3"
+---
+
+Apache Camel 4 is on the way for 1st half in 2023.
+
+The need for Camel 4 is mainly driven by Java open source projects migrating from `javax` to `jakarta` APIs
+and to keep up with popular runtimes such as Spring Boot and Quarkus.
+
+## Primary Goals 
+1. Migrate from `javax` -> `jakarta` (JEE 10)
+2. Java 17 as minimum
+3. Spring Framework 6
+4. Spring Boot 3
+5. Quarkus 3
+
+## Release Goals
+6. Release only what is ready (JEE10 / Java17)
+   This means that Camel components that are not ready (yet) will be dropped in a release until they are ready.
+7. Release Camel Core + Camel Spring Boot together
+8. Move Camel Karaf to Apache Karaf as karaf-camel sub-project 
+
+## Major Goals
+9. Support Java 17 features such as records, multiline strings, etc.
+10. EIP model without JAXB dependency (if possible)
+11. Endpoint URI parsing (do not use `java.net.URI`)
+12. Deprecate `message.getIn()` use `getMessage()` instead
+13. Deprecate/Remove camel-cdi
+14. Deprecate/Remove MDC logging (complex and buggy and does not fit modern app development)
+
+## Minor Goals
```

Comment:
I think we should be a bit more clear about the release, major and minor goals. I mean, in the sense of: 
* Minor goals: "we may try to do this, but we will drop this goal if we can't make it to the 4.0 release". 
* Major goals: "we won't block the release for this, but this will change/be removed at some point in 4.x release".
* Release goals: "this are non-negotiable release goals and we will block the release until this is complete". 

IMHO, this could give users a bit more clarity about what to expect.

Reply:
Granted: it is clarified for some of them, I'm just saying we should make it clear it applies to the others as well.